lib/connection_string.js

```diff
@@ -427,7 +427,7 @@ function parseQueryString(query, options) {
 
   // special cases for known deprecated options
   if (result.wtimeout && result.wtimeoutms) {
-    delete result.wtimeout;
+    result.wtimeout = undefined;
```

Comment:
I know this is counter to what I've said previously, but in these non-critical paths I think there's actual value to outright removing a known bad value. In the future we might solve this by "plucking" all the values we care about into a new object.